BeeSwift/Gallery/GalleryViewController.swift

```diff
@@ -431,9 +414,19 @@ class GalleryViewController: UIViewController, UICollectionViewDelegateFlowLayou
     }
     
     func numberOfSections(in collectionView: UICollectionView) -> Int {
-        return 1
+        switch (filteredGoals.isEmpty, goals.isEmpty) {
```

Comment:
`numberOfSections` seems like it should be a pure function which just returns a number and doesn't have side effects like updating the background. Maybe `updateGoals()` is a better place for this to live?

Reply:
I am not sure I understand. With this change the background is set according to the number of sections and is being set in the numberOfSections. So yeah, that method has the side effect of setting the background. 
As is, `updateGoals()` does not seem to be an ideal spot for the change either.
I made a change; take a look